### ISSUE TITLE
Consider to replace gulp-webserver with browsersync.io

### ISSUE THREAD

**Issue by StfBauer** [OfficeDev/generator-office#119]
Replacing the gulp-webserver with browsersync.io allows testing of multiple browser at the same time. By replacing gulp-webserver it is also possible to implement automatic reload of the app frame when files was changed. I think it would be pretty useful and this makes the refresh of the browser window obsolete. 
I currently tested it in an instance of the generator and from my point of view it worked pretty well.

``` javascript
var browserSync = require('browser-sync');
var reload = browserSync.reload;

gulp.task('serve-static', function () {
    browserSync({
        notify: false,
        port: 8443,
        server: {
            baseDir: ['.tmp', '.'],
            routes: {
                '/bower_components': 'bower_components',
                '/tmp': 'app/.tmp'
            }
        },
        https: true
    });
    gulp.watch([
      './**/*.js',
      './**/*.html',
      './**/*.css'])
      .on('change', reload);
});
```

If you like I can implement this.

**Comment by andrewconnell** [comment]
Can it implement HTTPS? The thing is that **gulp-webserver** includes a self-signed cert that can be registered just one time.

**Comment by StfBauer** [comment]
Yes it works the same way. Self signed cert is included in too,. Defined by the https switch. Plus you have more advanced stuff like socket.io for updates caused by code changes. Map bower components to /bower_components and so on

**Comment by andrewconnell** [comment]
Works for me… if you do this, can you also make sure all docs are updated that explain the process of trusting the cert & how to setup the auto reload? Love browser sync so this would be a good enhancement.

**Comment by StfBauer** [comment]
Sure. I will check for references but in general the procedure should be the same.

**Comment by StfBauer** [comment]
Added [pull request](https://github.com/OfficeDev/generator-office/pull/123) that contains the replacement of gulp-webserve with browsersync.io and minor fixes. See description of pull request.

**Comment by jthake** [comment]
nicely done sir. @andrewconnell did you want to validate this one ?

**Comment by andrewconnell** [comment]
The build broke… seems lots of tests are now failing with this PR:
https://travis-ci.org/OfficeDev/generator-office/builds/101673508

**Comment by StfBauer** [comment]
Seems like it is caused by the greater than version 2.1.1 reference of browser-sync in the _package.json files.
Should I change this to a specific version and re-issue pull request again?

**Comment by StfBauer** [comment]
Another issue might is the browser-sync compilation during installation.

**Comment by andrewconnell** [comment]
I’m not able to dig into it at this moment because I’m neck deep in other work. The tests should pass without the fix being removing them ;)

**Comment by StfBauer** [comment]
Found the issues and fixed them ;)

**Comment by StfBauer** [comment]
@andrewconnell Currently working on a blog post about the browser sync enhancements in yeoman generator. Mind if I create an additional page for browser-sync features to the documentation too?
I list there things like:
* Network throttling
* Cross browser testing
* Testing across multiple devices on the same network
* Live reload

**Comment by andrewconnell** [comment]
Sounds good... go for it. I'm not very involved in this project right now so I'd direct questions like this to @jthake instead...

**Comment by beth-panx** [comment]
@StfBauer I am working on the next version of generator-office now and decided to go with browsersync. Would love to have you code review some my work if possible. I will let you know once I have it on the magical interweb...

**Comment by StfBauer** [comment]
@beth-panx Awesome!!! If you need something let me know cause I already implemented that kind of stuff in yo office.

**Comment by beth-panx** [comment]
@StfBauer Browsersync has been implemented in v 1.0. Feedback wanted and more than welcomed!

**Comment by StfBauer** [comment]
Just need to find time to check it out. Maybe next week or so.
